### ISSUE TITLE
[enh] Make cluster definitions respect ROI boundaries 

### ISSUE THREAD

**Issue by nicholst** [NISOx-BDI/SwE-toolbox#140]
As extension to #41, change definition of a cluster to be constrained to lie within a parcel as based on ROI atlas provided for the surface/volume. This matches the behavior of some Connectome Workbench tools.

**Comment by nicholst** [comment]
There are standard files for the 'default' labels, but someone may want to use others.

https://github.com/Washington-University/HCPpipelines/tree/master/global/templates/91282_Greyordinates

* `91282_Greyordinates.dscalar.nii`  -  A brainordinates file with 19 subcortical/cerebellar regions, and it would seem 60 regions per hemisphere (just guessing as maximum value is 60)
* `Atlas_ROIs.2.nii.gz` - I assume this is the same as you could extract from the Greyordinates file, but in a convient 91x109x91 MNI space NIFTI (it has 19 unique values + 0's)
* `L.atlasroi.32k_fs_LR.shape.gii` & `R.atlasroi.32k_fs_LR.shape.gii` - The geometry for the CIFTI surfaces

@BryanGuillaume - Does this look like it's enough to go on?

Thoughts on how to proceed? I'm thinking that we *just* use these parcels in a loop... i.e. instead of submitting an entire surface to the bw_label function, we loop through all the parcels, zeroing out any values outside the current parcel, and then call bw_label on the 1-parcel-masked surface/volume.

Not very elegant, but should get the job done.

**Comment by nicholst** [comment]
Clarification from Tim Coalson:  The CIFTI format for subcortical/cerebellar/brainstem data has the regional definitions 'baked in', as suggested by the output of `wb_command -file-information` below.

Question for @BryanGuillaume: If we were to implement boundary-preserving clusters, is it clear how to use this information?

For the surface it simply must be the case that they will have a surface with integer labels that define each of the ROIs.

```
    AccumbensLeft:              135 voxels
    AccumbensRight:             140 voxels
    AmygdalaLeft:               315 voxels
    AmygdalaRight:              332 voxels
    BrainStem:                  3472 voxels
    CaudateLeft:                728 voxels
    CaudateRight:               755 voxels
    CerebellumLeft:             8709 voxels
    CerebellumRight:            9144 voxels
    DiencephalonVentralLeft:    706 voxels
    DiencephalonVentralRight:   712 voxels
    HippocampusLeft:            764 voxels
    HippocampusRight:           795 voxels
    PallidumLeft:               297 voxels
    PallidumRight:              260 voxels
    PutamenLeft:                1060 voxels
    PutamenRight:               1010 voxels
    ThalamusLeft:               1288 voxels
    ThalamusRight:              1248 voxels
```

**Comment by BryanGuillaume** [comment]
For the volume, I already know how to extract the ROI information. Actually, the function [swe_read_cifti_info](https://github.com/NISOx-BDI/SwE-toolbox/blob/master/swe_read_cifti_info.m) has already this ability. Though I would need to add some code to use this and create boundary-preserving clusters. 

For the surfaces, I am currently looking if I can extract this type of information

**Comment by BryanGuillaume** [comment]
For the surfaces, is this information also baked in the CIFTI or this is only present in [LR].atlasroi.32k_fs_LR.shape.gii files?

**Comment by BryanGuillaume** [comment]
From the CIFTI files I have, I can see 21 brain models, 2 for surfaces and 19 for volumes (one for each ROI), which allows me to get the volume ROI information. Nevertheless, from the surface brain models, I do not seem able to extract surface ROI. However, this info might be somewhere else in the XML.

**Comment by BryanGuillaume** [comment]
I have just found the answer in Tim Coalson's email: "no parcellation is built-in for surfaces.". Thus, the user will need to specify 2 GIfTI files encoding the surface ROI to make this work.

**Comment by nicholst** [comment]
Based on feedback from Mike Harms, the HCP 'Workbench' tools *only* respect subcortical boundaries.  Hence, for now, let's not worry about providing an option to respect cortical boundaries _unless_ you find it's simply easier to implement for both volume and surface together.

**Comment by BryanGuillaume** [comment]
I believe this issue has been solved in PR #149. Thus I am closing it.
